src/Orchard/Localization/Services/DefaultLocalizedStringManager.cs

```diff
@@ -118,70 +121,49 @@ public class DefaultLocalizedStringManager : ILocalizedStringManager {
         private IDictionary<string, string> LoadTranslationsForCulture(string culture, AcquireContext<string> context) {
             IDictionary<string, string> translations = new Dictionary<string, string>();
             string corePath = string.Format(CoreLocalizationFilePathFormat, culture);
-            string text = _webSiteFolder.ReadFile(corePath);
-            if (text != null) {
-                _localizationStreamParser.ParseLocalizationStream(text, translations, false);
-                if (!DisableMonitoring) {
-                    Logger.Debug("Monitoring virtual path \"{0}\"", corePath);
-                    context.Monitor(_webSiteFolder.WhenPathChanges(corePath));
-                }
-            }
+            LoadTranslationsFromLanguageFilePath(corePath, translations, context, false);
 
             foreach (var module in _extensionManager.AvailableExtensions()) {
                 if (DefaultExtensionTypes.IsModule(module.ExtensionType)) {
                     string modulePath = string.Format(ModulesLocalizationFilePathFormat, module.VirtualPath, culture);
-                    text = _webSiteFolder.ReadFile(modulePath);
-                    if (text != null) {
-                        _localizationStreamParser.ParseLocalizationStream(text, translations, true);
-
-                        if (!DisableMonitoring) {
-                            Logger.Debug("Monitoring virtual path \"{0}\"", modulePath);
-                            context.Monitor(_webSiteFolder.WhenPathChanges(modulePath));
-                        }
-                    }
+                    LoadTranslationsFromLanguageFilePath(modulePath, translations, context, true);
                 }
             }
 
-            foreach (var theme in _extensionManager.AvailableExtensions()) {
-                if (DefaultExtensionTypes.IsTheme(theme.ExtensionType) && _shellDescriptor.Features.Any(x => x.Name == theme.Id))
-                {
-
+            var currentTheme = _workContextAccessor.GetContext().CurrentTheme;
```

Comment:
this is the only functional change, splitting out the currently active theme and the inactive ones so inactive themes can be loaded then any overrides in the current theme will not be lost